mlos_bench/mlos_bench/schedulers/cyclic_scheduler.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""A simple single-threaded synchronous optimization loop implementation."""
+import logging
+from typing import Any, Dict, List
+from mlos_bench.environments.base_environment import Environment
+from mlos_bench.optimizers.base_optimizer import Optimizer
+from mlos_bench.schedulers.sync_scheduler import SyncScheduler
+from mlos_bench.storage.base_storage import Storage
+from mlos_bench.util import merge_parameters
+
+_LOG = logging.getLogger(__name__)
+
+
+class CyclicScheduler(SyncScheduler):
+    """
+    A simple single-threaded synchronous benchmarking loop implementation
+    that cycles through a list of provided configuration IDs.
+
+    This scheduler does not use any optimizer and is mainly used for benchmarking
+    specific configurations against each other only.
+
+    `cycle_config_ids` is the  list of existing configuration IDs to cycle through.
+    It can be provided via the scheduler's config, or overwritten through the global config.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        config: Dict[str, Any],
+        global_config: Dict[str, Any],
+        environment: Environment,
+        optimizer: Optimizer,
+        storage: Storage,
+        root_env_config: str,
+    ):
+        """
+        Create a new instance of the scheduler. The constructor of this and the derived
```

Comment:
Do we need this comment?  Since it's in the parent class, maybe remove it to avoid extra maintenance issues, unless there's something specific to this one.